src/components/UpdateButton/UpdateVisualizationContainer.js

```diff
@@ -1,64 +1,40 @@
-import { connect } from 'react-redux'
-import { tSetCurrentFromUi } from '../../actions/current.js'
+import { useSelector, useDispatch } from 'react-redux'
+import { tSetCurrentFromUi, acClearCurrent } from '../../actions/current.js'
 import {
     acSetLoadError,
     acClearLoadError,
     acSetVisualizationLoading,
 } from '../../actions/loader.js'
 import { acSetShowExpandedLayoutPanel } from '../../actions/ui.js'
 import { genericClientError } from '../../modules/error.js'
-import { validateLayout } from '../../modules/layoutValidation.js'
-import { sGetCurrent, sGetCurrentFromUi } from '../../reducers/current.js'
+import {
+    layoutHasProgramId,
+    validateLayout,
+} from '../../modules/layoutValidation.js'
+import { sGetCurrentFromUi } from '../../reducers/current.js'
+
+const UpdateVisualizationContainer = ({ renderComponent }) => {
+    const dispatch = useDispatch()
+    const currentFromUi = useSelector(sGetCurrentFromUi)
```

Comment:
You had commented previously to just use `useSelector(sGetCurrentFromUi)` directly instead of creating this variable. (I assume this can be done within a try/catch block). Did you still want to do that?

Reply:
That's right, previously it was only use in one place so the `const` wasn't needed, but as its now being used in two places instead of only one I think it makes sense to assign it to a `const`